packages/scheduler/npm/index.js

```diff
@@ -1,6 +1,8 @@
 'use strict';
 
-if (process.env.NODE_ENV === 'production') {
+if (typeof global.nativeRuntimeScheduler !== 'undefined') {
+  module.exports = global.nativeRuntimeScheduler;
```

Comment:
We can't be introducing `global` here because it doesn't exist in all environments.

Reply:
Before I dive into package.json and export conditionals. Is adding a null check for `global` on this line a bad pattern? 

Chaining it to the following:
```
if (global && typeof global.nativeRuntimeScheduler !== 'undefined')
```

Reply:
It can't be `if (global` because this is a ReferenceError. You want `if (typeof global !== 'undefined'` in this case. However, this is also a can of worms because some environments will bundle a `global` polyfill when they see this code. I think we need to take a step back and discuss in which specific cases we want to do it. It might just be easier to check `if (typeof nativeRuntimeScheduler !== 'undefined')` which effectively checks the global.

*However*, even if we do this, I'm not sure we should be doing it in this file. It seems a bit weird to check this for web open source users (even once). It prevents browsers from ever adding a global called `nativeRuntimeScheduler` (which might be ok but still). And this `npm/index.js` file is mostly the same between all packages so it seems strange to me we're adding a special case here. It's very easy to miss.